File: Singleton 單例模式/TypeScript/EagerSingleton.ts
class EagerSingleton {
  // 一開始就先建立實例
  private static instance: EagerSingleton = new EagerSingleton();

  // 私有化建構子
  // 避免在外面被建構出來
  private constructor() {}

  // 只能透過這邊取得實例
  public static getInstance(): EagerSingleton {
    return EagerSingleton.instance;
  }
}

File: Singleton 單例模式/Java/EagerSingleton.java
public class EagerSingleton {
    // 一開始就先建立實例
    private static EagerSingleton instance = new EagerSingleton();

    // 私有化建構子
    // 避免在外面被建構出來
    private EagerSingleton(){}

    // 只能透過這邊取得實例
    public static EagerSingleton getInstance(){
        return instance;
    }
}
